eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/permission/fiveway/FiveWayPermission.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.permission.fiveway;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.LocalBroadcastManager;
+
+import com.schedjoules.eventdiscovery.framework.permission.Permission;
+import com.schedjoules.eventdiscovery.framework.permission.PermissionRequestCallback;
+import com.schedjoules.eventdiscovery.framework.permission.broadcast.PermissionRequestResult;
+import com.schedjoules.eventdiscovery.framework.permission.broadcast.PermissionResultBroadcast;
+
+import static com.schedjoules.eventdiscovery.framework.permission.fiveway.FiveWayPermissionStatus.NOT_IN_MANIFEST;
+
+
+/**
+ * {@link Permission} supporting {@link FiveWayPermissionStatus}-es.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class FiveWayPermission implements Permission<FiveWayPermissionStatus>
+{
+
+    private static final int REQUEST_CODE = 45417;
+
+    private final Activity mActivity;
+    private final String mPermissionName;
+
+
+    public FiveWayPermission(Activity activity, String permissionName)
+    {
+        mActivity = activity;
```

Comment:
I've looked into it now and one benefit of moving them to parameters could be that `status()` only needs `Context` normally, so it could be used from a `Service` as well, but unfortunately `FiveWay` implementation uses `Activity` for `status()` too, to check for the `shouldShowRequestPermissionRationale` state. So what I can do is this:

```
interface Permission<S extends PermissionStatus, C extends Context>
{
    S status(C context);
    void request(Activity activity, PermissionRequestCallback<S> callback);
}

class FiveWayPermission implements Permission<FiveWayPermissionStatus, Activity>
```
So `FiveWayPermission` could only be used from `Activity` context.
This change would also mean we need to save `Activity/Context` references in components where we want to use `Permission`, like in current `PermissionProxy`.
But the possibility to use a `Permission` from non-Activity context, I think justifies this change, so I say we should change it as above. What do you say?